Src/Sarif.PatternMatcher.Sdk/ValidatorBase.cs

```diff
@@ -34,6 +34,9 @@ public abstract class ValidatorBase
             new Regex($@"The underlying connection was closed: Could not establish " +
                          "trust relationship for the SSL/TLS secure channel.", s_options);
 
+        private static readonly object sync = new object();
+        private static HttpClient httpClient;
```

Comment:
>httpClient [](http://example.com/codeflow?start=34&length=10)

why aren't we caching a per-thread instance of http client? with this solution, we wouldn't require locking, as all per-thread usage would be synchronous

Reply:
Changed to threadstatic.